project2-beat-that/script.js

```diff
@@ -1,4 +1,299 @@
+
+// init gameState 
+var gameState = `INTRODUCTION`; 
+
+var playerOne = [];
+var playerTwo = [];
+var playerOneCombine = []; 
+var playerTwoCombine = []; 
+
+var playerOneRollDice = function(){
+  var playerNumber = `1`;
+  counter = 0; 
+  while (counter < 2){
+    playerOne.push(rollDice());
+    counter +=1; 
+  }
+
+  console.log(`Player One arrary is ${playerOne}`);
+  var outputMessage = `Welcome Player ${playerNumber}. <br> 
+  Your first number is: ${playerOne[0]}. <br> 
+  Your second number is ${playerOne[1]}. <br> 
+  Choose the order of your dice. <br>  <br>  Enter SCORE to check current leader. 
+  <br>  <br> Enter LOWEST to change to lowest combined number mode. `
+  return outputMessage;
+}
+
+var playerTwoRollDice = function(){
+  var playerNumber = `2`;
+  counter = 0; 
+  while (counter < 2){
+    playerTwo.push(rollDice());
+    counter +=1; 
+  }
+ 
+  console.log(`Player Two arrary is ${playerTwo}`);
+  var outputMessage = `Welcome Player ${playerNumber}. <br> 
+  Your first number is: ${playerTwo[0]}. <br> 
+  Your second number is ${playerTwo[1]}. <br> 
+  Choose the order of your dice.<br>  
+  `
+  return outputMessage;
+}
+
+var introMessage = function(){
+  console.log(`Game state = ${gameState}`);
+  gameState == ROLL_DICE; 
+  output = `Press submit to start game. <br> `; 
+  return output; 
+}
+
+
+
+var main = function (input) { 
+  if (gameState == INTRODUCTION){
+    if (input == `SCORE`){
+      gameState = SCORE;
+      console.log(`Game state = ${gameState}`);
+    }
+    // else if (input == `LOWEST`){
+    //   gameState = LOWEST_COMBINE_NUMBDER;
+    //   console.log(`Game state = ${gameState}`);
+    
+    // }
+    else {
+      gameState = ROLL_DICE; 
+    }
+    return introMessage(); 
+  }
+  
+  // PLAYER 1 STARTS
+  if (gameState == ROLL_DICE){
+    if (input == `SCORE`){
+      gameState = SCORE;
+      console.log(`Game state = ${gameState}`);
+    }
+    // else if (input == `LOWEST`){
+    //   gameState = LOWEST_COMBINE_NUMBDER;
+    //   console.log(`Game state = ${gameState}`);
+    // }
+    else{
+      gameState = CHOOSE_DICE_ORDER;
+    }
+    
+    
+    return playerOneRollDice(); 
+  }
+  
+  // Player 1 chooses dice order 
+  if (gameState == CHOOSE_DICE_ORDER){
+    console.log(`New game state: ${gameState}`);
+    if (input == '1'){
+      playerOneNumber = String(playerOne[0]) + String(playerOne[1]);
+      playerOneCombine.push(playerOneNumber);
+      gameState = CHANGE_TO_PLAYER_2;
+      output = `Player 1 number is ${playerOneNumber}. <br> 
+      Player 2 please press submit to generate your numbers `
+      return output;
+    }
+    else if (input == '2'){
+      playerOneNumber = String(playerOne[1]) + String(playerOne[0]);
+      playerOneCombine.push(playerOneNumber);
+      gameState = CHANGE_TO_PLAYER_2;
+      output = `Player 1 number is ${playerOneNumber}. <br> 
+      Player 2 please press submit to generate your numbers `
+      return output;
+    }
```

Comment:
again repeated logic, see how in these two conditions, only line 114 and line 122 differ?

the rest are the same. you can have the other lines outside of these two conditions and only write them once